File: examples/typescript/events/list-event-destinations.ts
/**
 * List every event destination in the account.
 *
 *   STEDI_API_KEY=<api-key> tsx list-event-destinations.ts
 *
 * Pages until the token runs out; creates and deletes nothing.
 */

import { Stedi, type EventDestinationsDestinationSummary } from "@stedi/sdk";

const apiKey = process.env["STEDI_API_KEY"];
if (!apiKey) {
  throw new Error("STEDI_API_KEY is not set");
}

const client = new Stedi({ apiKey: { apiKey } });

const destinations: EventDestinationsDestinationSummary[] = [];
let pageToken: string | undefined;
let pages = 0;

do {
  const page = await client.listEventDestinations({ pageSize: 25, pageToken });
  destinations.push(...(page.items ?? []));
  pageToken = page.nextPageToken;
  pages += 1;
} while (pageToken);

console.log(`${destinations.length} destination(s) across ${pages} page(s):`);
for (const destination of destinations) {
  console.log(`  ${destination.id} [${destination.status}] ${destination.name}`);
  console.log(`    ${destination.destinationUrl} — ${destination.eventTypes?.join(", ")}`);
}

File: examples/python/events/list_event_destinations.py
# /// script
# requires-python = ">=3.12"
# dependencies = ["stedi==1.0.0"]
# ///

"""List every event destination in the account.

    STEDI_API_KEY=<api-key> uv run --script list_event_destinations.py

Pages until the token runs out; creates and deletes nothing.
"""

import asyncio
import os

from stedi import Config, Stedi
from stedi.models import (
    EventDestinationsDestinationSummary,
    ListEventDestinationsInput,
)


async def main(api_key: str) -> None:
    destinations: list[EventDestinationsDestinationSummary] = []
    pages = 0

    async with Stedi(Config(api_key=api_key)) as client:
        page_token: str | None = None
        while True:
            page = await client.list_event_destinations(
                ListEventDestinationsInput(page_size=25, page_token=page_token),
            )
            destinations.extend(page.items or [])
            pages += 1
            page_token = page.next_page_token
            if not page_token:
                break

    print(f"{len(destinations)} destination(s) across {pages} page(s):")
    for destination in destinations:
        print(f"  {destination.id} [{destination.status}] {destination.name}")
        print(f"    {destination.destination_url} — {', '.join(destination.event_types or [])}")


api_key = os.environ.get("STEDI_API_KEY")
if not api_key:
    raise SystemExit("STEDI_API_KEY is not set")

asyncio.run(main(api_key))
